drivers/include/periph/i2c.h

```diff
@@ -120,165 +170,177 @@ typedef enum {
     I2C_SPEED_HIGH,         /**< high speed mode:   ~3.4Mbit/s */
 } i2c_speed_t;
 #endif
+/** @} */
 
 /**
- * @brief   Initialize an I2C device to run as bus master
+ * @brief   I2C transfer flags
+ * @{
+ */
+#ifndef HAVE_I2C_FLAGS_T
+typedef enum {
+    I2C_ADDR10  = 0x01,     /**< use 10-bit device addressing */
+    I2C_REG16   = 0x02,     /**< use 16-bit register addressing */
+    I2C_NOSTOP  = 0x04,     /**< do not issue a STOP condition after transfer */
+    I2C_NOSTART = 0x08      /**< skip START sequence, ignores address field */
+} i2c_flags_t;
+#endif
+/** @} */
+
+/**
+ * @brief   I2C transfer results
+ */
+enum {
+    /**
+     * @brief   All bytes were transferred successfully
+     */
+    I2C_ACK = 0,
+    /**
+     * @brief   NACK when transferring the address byte
+     *
+     * After the address + the read/write bit were send, we got an NACK as
+     * response. This means most probably, that there is no slave with the used
+     * address on the bus, or the slave did just not respond for some reason.
+     */
+    I2C_ADDR_NACK = -1,
+    /**
+     * @brief   NACK while writing data bytes
+     *
+     * The slave responded to a data byte written to it with a NACK.
+     */
+    I2C_DATA_NACK = -2,
+     /**
+      * @brief   Internal error
+      *
+      * This status code is returned, on any other internal error that might
+      * have occurred. Possible reasons are not supported modes (e.g. 10-bit
+      * addressing).
+      */
+    I2C_ERR = -3
+};
+
+/**
+ * @brief   Initialize the given I2C bus
+ *
+ * The given I2C device will be initialized with the parameters as specified in
+ * the boards periph_conf.h, using the pins and the speed value given there.
  *
- * @param[in] dev           the device to initialize
- * @param[in] speed         the selected bus speed
+ * The bus MUST not be acquired before initializing it, as this is handled
+ * internally by the i2c_init function!
+ *
+ * @param[in] dev       the device to initialize
  *
  * @return                  0 on successful initialization
  * @return                  -1 on undefined device given
- * @return                  -2 on unsupported speed value
  */
-int i2c_init_master(i2c_t dev, i2c_speed_t speed);
```

Comment:
Why did we drop the speed param here ?

Reply:
Because we found that it is not actually needed in the interface and it is more efficient to statically configure this value in the board configuration. As I2C is not like SPI, where each device on the bus can be interfaces with a different clock speed, the bus clock for I2C buses needs to be assigned globally for all devices on the bus, so we might as well define it as part of the I2C config in the boards `periph_conf.h`.

Per default, the `i2c_init_master()` is to be called during system initialization (same as `spi_init()` in from `periph_init()`). In this context, it is much easier to pass the clock speed as configuration parameter instead of a function parameter...